core/camel-base/src/main/java/org/apache/camel/impl/engine/AbstractCamelContext.java

```diff
@@ -2757,6 +2810,7 @@ protected void doStartCamel() throws Exception {
         ServiceHelper.startService(lifecycleStrategies);
         for (LifecycleStrategy strategy : lifecycleStrategies) {
             try {
+                strategy.onContextStarting(this);
                 strategy.onContextStart(this);
```

Comment:
Is this the old name onContextStart? And should it have been deprecated or what?

Reply:
yes the old name is `onContextStart` and I have deprecated it

Reply:
Perfect